pkg/reconciler/v1alpha1/taskrun/resources/pod_test.go

```diff
@@ -435,3 +420,42 @@ func TestMakeWorkingDirScript(t *testing.T) {
 		})
 	}
 }
+
+func TestAddReadyAnnotation(t *testing.T) {
+	type testcase struct {
+		desc                       string
+		pod                        *corev1.Pod
+		updateFunc                 UpdatePod
+		expectedAnnotationsContain map[string]string
+		expectedErr                error
+	}
+	testerror := xerrors.New("error updating pod")
+	for _, c := range []testcase{{
+		desc:                       "missing ready annotation is added to provided pod",
+		pod:                        &corev1.Pod{},
+		updateFunc:                 func(p *corev1.Pod) (*corev1.Pod, error) { return p, nil },
+		expectedAnnotationsContain: map[string]string{ReadyAnnotation: readyAnnotationValue},
+		expectedErr:                nil,
+	}, {
+		desc:                       "errors experienced during update are returned",
+		pod:                        &corev1.Pod{},
+		updateFunc:                 func(p *corev1.Pod) (*corev1.Pod, error) { return p, testerror },
+		expectedAnnotationsContain: map[string]string{ReadyAnnotation: readyAnnotationValue},
+		expectedErr:                testerror,
```

Comment:
instead of using `expectedErr` i find it much clearer to have two completely different tests, one for success conditions and one for error conditions (less potential for something to go wrong and also easier to read)

Reply:
I've separated this into two tests now, one for the successful case and one for the error case.